src/xinterpreter.cpp

```diff
@@ -240,15 +240,60 @@ namespace xpyt
     void interpreter::redirect_display()
     {
         py::module sys = py::module::import("sys");
+        py::module py_json = py::module::import("json");
 
         py::module xeus_python_display = py::module::import("xeus_python_display");
         m_displayhook = xeus_python_display.attr("XPythonDisplay")();
 
-        py::cpp_function publish_display = [this](int execution_counter, py::object obj){
+        py::cpp_function publish_display = [this, py_json](int execution_counter, py::object obj){
             if (!obj.is_none())
             {
                 xeus::xjson pub_data;
-                pub_data["text/plain"] = py::str(obj);
+
+                if (hasattr(obj, "_repr_mimebundle_"))
+                {
+                    pub_data = xeus::xjson::parse(static_cast<std::string>(
+                        py::str(py_json.attr("dumps")(obj.attr("_repr_mimebundle_")()))
+                    ));
+                }
+                else if (hasattr(obj, "_repr_html_"))
+                {
+                    pub_data["text/html"] = static_cast<std::string>(
+                        py::str(obj.attr("_repr_html_")())
+                    );
+                }
+                else if (hasattr(obj, "_repr_json_"))
+                {
+                }
+                else if (hasattr(obj, "_repr_jpeg_"))
+                {
+                    pub_data["image/jpeg"] = static_cast<std::string>(
+                        py::str(obj.attr("_repr_jpeg_")())
+                    );
+                }
+                else if (hasattr(obj, "_repr_png_"))
+                {
+                    pub_data["image/png"] = static_cast<std::string>(
+                        py::str(obj.attr("_repr_png_")())
+                    );
+                }
+                else if (hasattr(obj, "_repr_svg_"))
+                {
+                    pub_data["image/svg+xml"] = static_cast<std::string>(
+                        py::str(obj.attr("_repr_svg_")())
+                    );
+                }
+                else if (hasattr(obj, "_repr_latex_"))
+                {
+                }
+                else if (hasattr(obj, "__repr__"))
+                {
+                    pub_data["text/plain"] = static_cast<std::string>(
+                        py::str(obj.attr("__repr__")())
+                    );
+                }
```

Comment:
Should we check if it has `ipython_display` hook and say that it is not supported with a warning?

Reply:
Maybe, is `_ipython_display_` something that is used a lot? Is it used in widget libraries? I wouldn't want to send a warning every single time the user wants to display widgets.

Reply:
Ok I have my answer for widgets, I guess we can display a warning